### ISSUE TITLE
500 instead of 404 when no homepage is defined in production

### ISSUE THREAD

**Issue by greg0ire** [symfony/symfony#27945]
**Symfony version(s) affected**: 4.1.1

**Description**  
I'm working on a microservice that does not have a homepage, because we have this policy of prefixing our APIs with `/api`. I didn't configuration a homepage, and while in dev, I get a welcome page aimed at developers, in production, I get a 500 because this welcome page is displayed by throwing a `NoConfigurationException`

Here is what I believe is the configuration responsible for this:

https://github.com/symfony/symfony/blob/adb137d5c1f5c2b0831b5135c0e171b46bac81ca/src/Symfony/Component/Routing/Matcher/Dumper/PhpMatcherDumper.php#L198

**How to reproduce**  
```
cd /tmp
composer create-project symfony/skeleton
cd skeleton
sed -i 's/dev/prod/' .env
php -S 127.0.0.1:8000 -t public
```

```2018-07-13T15:05:48+00:00 [critical] Uncaught Exception: No route found for "GET /"
[Fri Jul 13 17:05:48 2018] PHP Fatal error:  Uncaught Symfony\Component\Routing\Exception\NoConfigurationException in /tmp/skeleton/var/cache/prod/srcProdProjectContainerUrlMatcher.php:65
Stack trace:
#0 /tmp/skeleton/var/cache/prod/srcProdProjectContainerUrlMatcher.php(21): srcProdProjectContainerUrlMatcher->doMatch('/', Array, Array)
#1 /tmp/skeleton/vendor/symfony/routing/Matcher/UrlMatcher.php(107): srcProdProjectContainerUrlMatcher->match('/')
#2 /tmp/skeleton/vendor/symfony/routing/Router.php(262): Symfony\Component\Routing\Matcher\UrlMatcher->matchRequest(Object(Symfony\Component\HttpFoundation\Request))
#3 /tmp/skeleton/vendor/symfony/http-kernel/EventListener/RouterListener.php(115): Symfony\Component\Routing\Router->matchRequest(Object(Symfony\Component\HttpFoundation\Request))
#4 /tmp/skeleton/vendor/symfony/event-dispatcher/EventDispatcher.php(212): Symfony\Component\HttpKernel\EventListener\RouterListener->onKernelRequest(Object(Symfony\Component\HttpKernel\Event\GetResponseEvent), 'kernel.request', Object(Symfony\Component\EventDispatcher\ in /tmp/skeleton/vendor/symfony/http-kernel/EventListener/RouterListener.php on line 139
[Fri Jul 13 17:05:48 2018] 127.0.0.1:51332 [500]: / - Uncaught Symfony\Component\Routing\Exception\NoConfigurationException in /tmp/skeleton/var/cache/prod/srcProdProjectContainerUrlMatcher.php:65
Stack trace:
#0 /tmp/skeleton/var/cache/prod/srcProdProjectContainerUrlMatcher.php(21): srcProdProjectContainerUrlMatcher->doMatch('/', Array, Array)
#1 /tmp/skeleton/vendor/symfony/routing/Matcher/UrlMatcher.php(107): srcProdProjectContainerUrlMatcher->match('/')
#2 /tmp/skeleton/vendor/symfony/routing/Router.php(262): Symfony\Component\Routing\Matcher\UrlMatcher->matchRequest(Object(Symfony\Component\HttpFoundation\Request))
#3 /tmp/skeleton/vendor/symfony/http-kernel/EventListener/RouterListener.php(115): Symfony\Component\Routing\Router->matchRequest(Object(Symfony\Component\HttpFoundation\Request))
#4 /tmp/skeleton/vendor/symfony/event-dispatcher/EventDispatcher.php(212): Symfony\Component\HttpKernel\EventListener\RouterListener->onKernelRequest(Object(Symfony\Component\HttpKernel\Event\GetResponseEvent), 'kernel.request', Object(Symfony\Component\EventDispatcher\ in /tmp/skeleton/vendor/symfony/http-kernel/EventListener/RouterListener.php on line 139
[1]    7903 segmentation fault (core dumped)  php -S 127.0.0.1:8000 -t public


```

**Possible Solution**  
Catch the exception somehwere?

**Comment by nicolas-grekas** [comment]
Duplicate of #25844, fixed as a new feature by #27519?

**Comment by greg0ire** [comment]
Yes, it works, thanks!
